docs/docsite/rst/network_aggregate_resources.rst

```diff
@@ -0,0 +1,175 @@
+
+* Should these vlans be in *addition* to the existing configuration?  - *additive*
+* Should these vlans be the *only* ones present? - *aggregate*
+
+
+.. versionadded:: 2.4
+
+   The ``aggregate:`` option has been added in Ansible 2.4 and is available in certain modules, see the modules documentation to see if the feature is available.
+
+Additive resources
+===================
+
+Continuing our vlan example, the following task ensure that vlans ``1``, ``2`` and ``3`` are in the states specified in the task, in `addition` to any existing vlans.
+
+This task *will not* change any vlans already configured on the switch, apart from the ones specified in the task. Ansible will ensure that the vlans specified in the task exist with the `name` and `state` specified.
+
+.. code-block:: yaml
+
+   - name: configure vlans neighbor in addition to existing
+     net_vlan:
+       vlan_id: “{{ item.vlan_id }}”
+       name: “{{ item.name }}”
+       state: “{{ item.state | default(‘active’) }}”
+     with_items:
+       - { vlan_id: 1, name: default }
+       - { vlan_id: 2, name: mgmt }
+       - { vlan_id: 3, state: suspend }
+
+FIXME Should we even document ``with_items``, is that just confusing the matter. Should this be the same as `Aggregate resources` though without ``purge: yes``?
+
+FIXME Only showing ``with_items`` to show how it could be done
+
+
+Would be better written as
+
+.. code-block:: yaml
+
+   - name: configure vlans neighbor (delete others)
+     net_vlan:
+       aggregate:
+         - { vlan_id: 1, name: default }
+         - { vlan_id: 2, name: mgmt }
+         - { vlan_id: 3, state: suspend }
+       state: active
+
+This task is very similar to the `additive resource` example above, though with the following differences:
+
+* The module runs only once, rather than n times as it does ``with_items``
+* There is no way to to use ``purge`` on ``with_items``
+
+
+Aggregate resources
+===================
+
+Consider the following example:
+
+.. code-block:: yaml
+
+   - name: configure vlans neighbor (delete others)
+     net_vlan:
+       aggregate:
+         - { vlan_id: 1, name: default }
+         - { vlan_id: 2, name: mgmt }
+         - { vlan_id: 3, state: suspend }
+       state: active
+       purge: yes # Important
+
+FIXME Descrbe ``purge``
+* The ``purge:`` option (which defaults to `no`) ensures that **only** the specified entries are present. All other entries will be **deleted**.
+
+
+FIXME: the ``state:`` is Local overrides of global module values
+
+.. warning:: Why does ``purge`` default to ``no``?
+
+   To prevent from accidental deletion ``purge`` is always set to ``no``. This requires playbook writers to add ``purge: yes`` to enable this.
+
+
+
+Local overrides of global module values
+=======================================
+
+* What
+* Why: Cleaner short hand. Allows you to separate out what's common from what's item specific.
```

Comment:
Please write this section out with full sentences. It reads like shorthand notes.